Improvements to repo manager

antis81 opened the pull request:
- [x] emit signals for (de-)activation of repository items

This one is needed by the topic branch in MacGitverModules/topic/RepoTree (PR 66).

antis81 committed:
emit a repositoryDeactivated signal when switching the active repository

scunz commented on libMacGitverCore/MacGitver/RepoManager.cpp:
```diff
@@ -207,8 +207,13 @@ void RepoManager::internalActivate( RepositoryInfo* repository )
+    if ( prev )
+        emit repositoryDeactivated( prev );
+
+    emit repositoryActivated( mActiveRepo );
```
Why this complex flow? :-)

```
if(mActiveRepo)
    emit repositoryDeactivated(mActiveRepo);
mActiveRepo = repository;
if(mActiveRepo) /* Omitting this was a design mistake along not creating a 'deactivate' in the first place */
    emit repositoryActivated(mActiveRepo);
```

antis81 replied:
Because of the assignment order :-). Nevertheless, I'm fine with that one either. For the second change, I was not sure at first, but it is obvious on a second glimpse. Taking this one into the PR.

scunz replied:
Omitting the `repositoryActivated(NULL)` _might_ cause trouble; but please do it anyway. It was a really bad mistake from me to not design this with a `deactivate` signal in the first place. However, there might be code out in the wild that explicitly expects this signal (or just works around getting it). **I will** take care of those as soon as this has gone into MGV.

antis81 replied:
Ok, it's in! Seems working so far. But, we never know ...

antis81 committed:
corrected call order of de-/activated signals in RepoMan

scunz commented:
Okay, I'm holding a merge back for now. I will check this code out and add any corrections possibly required for the `NULL`-Problem, later this evening.

scunz commented:
rebased this on on top of `development` in order to mingle it with the remaining development-branches.

scunz committed:
Introduce a new signal that kicks on activation

This new signal is triggered when a transition from NO-Active-Repo to
1-Active-Repo and vice versa happens.

scunz commented:
Actually, it looks like I was wrong on the complexness of the code. Actually, without my addition to this, the `RepoTree` code would have to keep track itself of the fact that there is a repository active.

scunz committed:
Teach RepoStateWidget to act on deactivation

scunz commented on libMacGitverCore/MacGitver/RepoManager.cpp:
```diff
@@ -200,15 +200,36 @@ void RepoManager::internalClosedRepo(RepositoryInfo* repository)
+        if(mActiveRepo)
+            emit repositoryActivated(mActiveRepo);
+
+        if ((mActiveRepo != NULL) != (old != NULL)) {
```
I think, putting it like this, makes it even more obvious what was meant. So I am confident to remove the comment completely.
